File: 02-week-2--recursion/01-day-1--recursive-counting/ruby/recursive_count.rb
def recursive_count(num = 0)
  return if num >= 10
  puts num
  recursive_count(num + 1)
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

# add a base case for the number being equal or less than 9 to return
# log out the number
# recurisvely call the added number

File: 02-week-2--recursion/01-day-1--recursive-counting/javascript/recursive_count.js
function recursiveCount(num = 0) {
  if (num >= 9){
    return console.log(num)
  }
  console.log(num)
  recursiveCount(num + 1)
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
//recursively call the function
// increase the counter each time the method is called
//stop recursion if it hits 9. counter starts at 0

//this works because i first had to write the conditional for the base case
// then return the function if the base case was hit
// if its not hit, then console log the number, and recursively call the number with an added 1